File: Algorithms/Implementations/sherlock_and_squares.c
#include <stdio.h>
#include <math.h>

int square_count(int a, int b)
{
    int count = 0;
    int sqrt_start = (int) ceil(sqrt(a));
    int sqrt_end = (int) floor(sqrt(b));
    return sqrt_end - sqrt_start + 1;
}

int main()
{
    int test_count;
    scanf("%d", &test_count);

    int tests[test_count];
    for(int i=0; i < test_count; i++)
    {
        int a, b;
        scanf("%d %d", &a, &b);
        printf("%d\n", square_count(a, b));
    }
    return 0;
}

File: Algorithms/Implementations/sherlock_and_squares.py
import math
test_count = int(raw_input().strip())
tests = []

def square_count(a, b):
    count = 0
    sqrt_start = a**(0.5)
    sqrt_end = b**(0.5)
    return int(math.floor(sqrt_end) - math.ceil(sqrt_start) + 1)

for i in xrange(test_count):
    a, b = raw_input().split()
    tests.append([int(a), int(b)])

for a, b in tests:
    print square_count(a, b)
